Sources/Queues/Application+Queues.swift

```diff
@@ -21,14 +21,25 @@ extension Application {
         }
 
         final class Storage {
+            enum QueuesType: String {
+                case `default` = "queues"
+                case inProcessJob = "inProcessJobsQueues"
+                case scheduledJobs = "scheduledJobsQueues"
+            }
             public var configuration: QueuesConfiguration
-            let command: QueuesCommand
+            private (set) var commands: [QueuesType: QueuesCommand]
             var driver: QueuesDriver?
 
             public init(_ application: Application) {
                 self.configuration = .init(logger: application.logger)
-                self.command = .init(application: application)
-                application.commands.use(self.command, as: "queues")
+                let command: QueuesCommand = .init(application: application)
```

Comment:
The access race error is most likely caused by the lack of a `Lock` here

Reply:
Thank's I have tried to add lock but it doesn't help. It can be related to `run` scheduled twice but I do not understand how that possible.

*QueueWorker*

```
 public func run() -> EventLoopFuture<Void> {
        self.queue.pop().flatMap { id in

```


*TestQueue*

```
    func pop() -> EventLoopFuture<JobIdentifier?> {
        return self.context.eventLoop.makeSucceededFuture(context.application.queues.test.queue.popLast())
    }
```

Not sure how but `run` seems to be executed twice.

Reply:
What happens if we make the previous comment an array and then rerun?